Applying Clang format

norihiro-w opened the pull request:
Same as we do in ogs6, I would like to suggest using clang format also in ogs5 except for external codes under ThirdParty dir. Having the consistent format is quite helpful, better code readability and less merge conflicts.

bilke commented:
I think we should wait with this until the changes from Kiel are merged (they are still working on it). Otherwise we have a hard time integrating them.

norihiro-w commented:
I don't think it makes merging harder, because they just need to apply clang-format to their code. Indeed it could make it even easier for them, because applying clang-format deletes all tiny differences between the versions, e.g. indentations, spaces at the end of lines, etc.

bilke commented:
Good points indeed. :+1:

norihiro-w commented:
TODO: check format in travis for incoming PRs

norihiro-w commented:
@bilke I added coding style check in travis. can you please take a look again? If a PR doesn't follow the code style, "case=codingstyle " in the travis test should fail  and its log includes a message `ERROR: Found formatting errors!`.

bilke commented:
@norihiro-w Good work! For me it is good but hopefully it is not too strict for other people..

bilke commented on .travis.yml:
```diff
@@ -1,47 +1,61 @@
+# Ubuntu 14.04 Trusty support
+sudo: required
```
Just a comment: Jobs requiring `sudo` [are slower](https://docs.travis-ci.com/user/ci-environment/) (see `Boot Time`) but otherwise you can not use `apt-get install`.

norihiro-w replied:
unfortunately OpenMPI and PETSc installation requires `apt-get install` because the packages are not in their white list (as far as I tested)

norihiro-w commented:
Any other comments on this? I'll wait until Wednesdays and then go for merging.

wenqing commented:
How to apply the format? Seems not automatically.

norihiro-w commented:
Run clang-format -style=file will do it. There are also plugins for many IDEs including Visual studio.

wenqing commented:
@norihiro-w I see. That's great.

norihiro-w commented:
UPDATE: 
- Modified the format style to minimize changes in current codes. 
- Added Linux and Windows versions of a batch script (under scripts/clang-format) applying clang-format to all source code files in OGS. Windows users can just double-click it to start the formatting.
- TODO: Since version 3.8, clang-format automatically sorts the order of header includes, which doesn't really work for OGS codes because of comment lines in-between #includes. To deactivate this, a user should add `SortIncludes:    false` in .clang-format.

norihiro-w commented:
About the formatting policy, I'm thinking in this way
- In general we ask developers to apply clang-format or change the code by hand before they submit a PR. Travis will tell us if the changes follow the format or not. Note that the travis script checks only the codes being changed and ignores already committed codes.
- For bug-fix or any urgent code changes, we will merge the commits even if the codes don't follow the format. Clang-format can be applied later.

norihiro-w commented:
will be merged after #33
